app/services/user_service.rb

```diff
@@ -0,0 +1,58 @@
+module UserService
+  def self.create(params:)
+    new_user = _user_from_params(params)
+
+    if new_user.performer?
+      User.transaction do
+        old_user = User.find_by(current_spot: new_user.current_spot)
+        deactivate_user(buck_id: old_user.buck_id) unless old_user.nil?
+      end
+    end
+
+    new_user.save! if new_user.valid?
+    new_user
+  end
+
+  def self.deactivate_user(buck_id:)
+    user = User.find(buck_id)
+    user.active = false
+    user.current_spot = nil
+    user.original_spot = nil
+    user.save!
+    user
+  end
+
+  # rubocop:disable Metrics/MethodLength
+  def self._user_from_params(params)
+    permitted = _create_params(params)
```

Comment:
I think this does what you are trying to do here while keeping within your method length limits:

```ruby
def self._user_from_params(params)
  user_attributes = _create_params(params).to_h.merge(password_digest: SecureRandom.base64)

  if user_attributes[:spot].present?
    row = user_attributes[:spot][:row]&.downcase&.to_sym
    file = user_attributes[:spot][:file]&.to_i
    spot = Spot.find_by(row: row, file: file)
    user_attributes.delete(:spot)
    user_attributes.merge(current_spot: spot, original_spot: spot)
  end

  User.new(user_attributes)
end
```